clients/vscode-hlasmplugin/scripts/syntaxes/hlasm_tmgrammar_template.txt

```diff
@@ -305,48 +362,82 @@
-      "match": "^\\.[A-Za-z@#$_][A-Za-z@#$_0-9]*(?<=^.{0,71})",
+      "match": "(?<=^${listingDetails}$)\\.[A-Za-z@#$_][A-Za-z@#$_0-9]*(?<=^${listingDetails}$.{0,71})",
       "name": "keyword.control.hlasm"
     },
     "sequenceSymbolOperandField": {
-      "match": "\\.[A-Za-z@#$_][A-Za-z@#$_0-9]*(?=,|\\s|$|(?<=^.{71}))(?<=^.{0,71})",
+      "match": "\\.[A-Za-z@#$_][A-Za-z@#$_0-9]*(?=,|\\s|$|(?<=^${listingDetails}$.{71}))(?<=^${listingDetails}$.{0,71})",
       "name": "keyword.control.hlasm"
     },
     "instruction": {
-      "match": "(?<=^\\S*)\\s+\\S+(?<=^.{0,71})",
+      "match": "(?<=^${listingDetails}$\\S*)\\s+\\S+(?<=^${listingDetails}$.{0,71})",
       "name": "entity.name.function.hlasm"
     },
     "remark": {
-      "match": "(\\S+)(?<=^.{0,71})",
+      "match": "(\\S+)(?<=^${listingDetails}$.{0,71})",
       "name": "comment.hlasm"
     },
     "operators": {
-      "match": "[=(),*/+-\\.](?<=^.{0,71})",
+      "match": "[=(),*/+-\\.](?<=^${listingDetails}$.{0,71})",
       "name": "keyword.operator.hlasm"
     },
     "numbers": {
-      "match": "(?<=[=(),*\\s+])-?\\d*((?=[=(),*+\\s]|$)(?<=^.{0,71}))",
+      "match": "(?<=[=(),*\\s+])-?\\d*((?=[=(),*+\\s]|$)(?<=^${listingDetails}$.{0,71}))",
       "name": "constant.numeric.hlasm"
     },
     "variable": {
-      "match": "&(([A-Za-z@#$_][A-Za-z@#$_0-9]*)|([(].*?[)]))(?<=^.{0,71})",
+      "match": "&(([A-Za-z@#$_][A-Za-z@#$_0-9]*)|([(].*?[)]))(?<=^${listingDetails}$.{0,71})",
       "name": "variable.other.readwrite.hlasm"
     },
     "ordinarySymbol": {
-      "match": "[A-Za-z@#$_][A-Za-z@#$_0-9]*(?<=^.{0,71})",
+      "match": "[A-Za-z@#$_][A-Za-z@#$_0-9]*(?<=^${listingDetails}$.{0,71})",
       "name": "support.class.hlasm"
     },
     "asmSpecialStatements": {
-      "match": "(?<=[=(),*+\\s])=?\\d*(A|B|C|D|E|F|G|H|P|V|X|Z)(L\\d*)?((?=[=(),*\\s+]|$)(?<=^.{0,71}))",
+      "match": "(?<=[=(),*+\\s])=?\\d*(A|B|C|D|E|F|G|H|P|V|X|Z)(L\\d*)?((?=[=(),*\\s+]|$)(?<=^${listingDetails}$.{0,71}))",
       "name": "punctuation.definition.tag.hlasm"
     },
     "continuationSymbol": {
-      "match": "\\s*(?<=^.{71})\\S",
+      "match": "\\s*(?<=^${listingDetails}$.{71})\\S",
       "name": "keyword.other.hlasm"
     },
     "sequenceNumbers": {
-      "match": "\\s*.?(?<=^.{72}).+$",
+      "match": "\\s*.?(?<=^${listingDetails}$.{72}).+$",
       "name": "invalid.ignored"
+    },
+    "listingSyntax": {
+      "match": "^${listingOffset}$\\S+\\s(\\S{4}\\s\\S{4})",
+      "captures": {
+        "1": {
+          "patterns": [
+            {
+              "include": "#objectCode"
+            }
+          ]
+        }
+      }
+    },
+    "objectCode": {
+      "patterns": [
+        {
+          "match": "(.{2})(.{7})",
+          "captures": {
+            "1": {
+              "name": "entity.name.function.hlasm"
+            },
+            "2": {
+              "name": "constant.numeric.hlasm"
+            }
+          }
+        }
+      ]
+    },
```

Comment:
maybe one small detail?
```suggestion
    "listingSyntax": {
      "match": "^${listingOffset}$\\S+\\s([0-3][0-9A-F]{3}|[4-9AB][0-9A-F]{3} [0-9A-F]{4}|[CDEF][0-9A-F]{3} [0-9A-F]{4} [0-9A-F]{4})",
      "captures": {
        "1": {
          "patterns": [
            {
              "include": "#objectCode"
            }
          ]
        }
      }
    },
    "objectCode": {
      "patterns": [
        {
          "match": "(.{2})(.*)",
          "captures": {
            "1": {
              "name": "entity.name.function.hlasm"
            },
            "2": {
              "name": "constant.numeric.hlasm"
            }
          }
        }
      ]
    },
```

Reply:
Sounds good! I'll try to add test cases for this as well

Reply:
This is how it looks now:
![image](https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/assets/99467904/018f1aef-9dda-4a21-acc4-3057fa4b0a76)